web/src/main/java/ca/hendriks/bartender/web/ingredient/IngredientsRestController.java

```diff
@@ -28,4 +29,18 @@ public Ingredient addIngredient(@RequestBody Ingredient ingredient) {
         return ingredientRepository.save(ingredient);
     }
 
+    @PutMapping
+    public void updateIngredient(@RequestBody List<String> ingredientChangeList){
```

Comment:
This could work... but then the REST client needs to know the exact order to send the attributes in... which is, awkward. Better to use a key-value map, or JSON.

eg. `PUT /ingredients/id/1?name=newName&type=newType`

or

eg. `PUT /ingredients/id/1`

```
{
   name: newName,
   type: newType
}
```

Reply:
Ok I like the map idea

Reply:
If you put the parameters in the URL, then some clients/servers limit you to 256 characters. More often than not, you will see a POST put the parameters in the body, and typically, that format is JSON.